ipc/provider/src/lib.rs

```diff
@@ -149,11 +168,53 @@ impl IpcProvider {
                 }
             }
             config::subnet::SubnetConfig::Fevm(_) => {
-                // TODO: add more checks later
+                // TODO: More checks to come
             }
         }
         Ok(())
     }
+
+    fn check_sender(
+        &mut self,
```

Comment:
making this `mut` will lead to methods that require sending `txn` to use `mut self`, which might force owner of this struct to use `mut self`, i.e. `fendermint` top down provider might encounter this issue. If `mut` is not necessary, we can just stick with `self`, else callers might have to use `RWLock` for this.

Reply:
Tracked in this issue for a follow-up: https://github.com/consensus-shipyard/ipc-agent/issues/315